File: src/main/java/com/rokucraft/rokubot/entities/Repository.java
package com.rokucraft.rokubot.entities;

import org.kohsuke.github.GHRepository;
import org.spongepowered.configurate.objectmapping.ConfigSerializable;
import org.spongepowered.configurate.objectmapping.meta.Required;

@ConfigSerializable
public record Repository(
        @Required String name,
        @Required String repositoryUrl
) {
    public static Repository of(GHRepository repo) {
        return new Repository(repo.getName(), repo.getHtmlUrl().toString());
    }
}

File: src/main/kotlin/com/rokucraft/rokubot/entities/Repository.kt
package com.rokucraft.rokubot.entities

import org.kohsuke.github.GHRepository
import org.spongepowered.configurate.objectmapping.ConfigSerializable
import org.spongepowered.configurate.objectmapping.meta.Required

@ConfigSerializable
data class Repository(
    @Required val name: String,
    @Required val repositoryUrl: String
) {
    companion object {
        fun of(repo: GHRepository): Repository {
            return Repository(repo.name, repo.htmlUrl.toString())
        }
    }
}
